### ISSUE TITLE
Cannot delete hibernated cluster after enabling APIServerSNI feature gate

### ISSUE THREAD

**Issue by ialidzhikov** [gardener/gardener#2887]
**How to categorize this issue?**

/area usability
/kind bug
/priority normal

**What happened**:
I triggered deletion for a Shoot which was hibernated since longtime. As part of the deletion flow, gardenlet wakes up the apiserver to ensure proper cleanup. However for such Shoot the apiserver fails to start with:
```
$ k -n shoot--foo--bar logs kube-apiserver-6d9c4f66bd-qplrj kube-apiserver
Error: invalid argument "" for "--advertise-address" flag: failed to parse IP: ""
Usage:
  kube-apiserver [flags]

Generic flags:

      --advertise-address ip                         The IP address on which to advertise the apiserver to members of the cluster. This address must be reachable by the rest of the cluster. If blank, the --bind-address will be used. If --bind-address is unspecified, the host's default interface will be used.
      --cloud-provider-gce-l7lb-src-cidrs cidrs      CIDRs opened in GCE firewall for L7 LB traffic proxy & health checks (default 130.211.0.0/22,35.191.0.0/16)
```

Checking the Services, I see that the kube-apiserver Service is still of type LoadBalancer:

```
$ k -n shoot--foo--bar get svc kube-apiserver
NAME             TYPE           CLUSTER-IP       EXTERNAL-IP                                                               PORT(S)         AGE
kube-apiserver   LoadBalancer   10.243.126.169   a2d5e2238dc424faeac0fb013da0afoo-1566849bar.eu-west-1.elb.amazonaws.com   443:30654/TCP   11h
```

Checking the kube-apiserver container command, I see that advertise-address flag is empty:

```
      containers:
      - command:
        - /usr/local/bin/kube-apiserver
        - --enable-admission-plugins=Priority,NamespaceLifecycle,LimitRanger,PodSecurityPolicy,ServiceAccount,NodeRestriction,DefaultStorageClass,DefaultTolerationSeconds,ResourceQuota,StorageObjectInUseProtection,MutatingAdmissionWebhook,ValidatingAdmissionWebhook
        - --admission-control-config-file=/etc/kubernetes/admission/admission-configuration.yaml
        - --allow-privileged=true
        - --anonymous-auth=false
        - --audit-log-path=/var/lib/audit.log
        - --audit-policy-file=/etc/kubernetes/audit/audit-policy.yaml
        - --audit-log-maxsize=100
        - --audit-log-maxbackup=5
        - --authorization-mode=Node,RBAC
        - --advertise-address=
        - --client-ca-file=/srv/kubernetes/ca/ca.crt
        - --enable-aggregator-routing=true
```

**What you expected to happen**:

I would expect to be able to delete a hibernated Shoot after enablement of APIServerSNI feature gate.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:

- Gardener version: master
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Others:

**Comment by vpnachev** [comment]
This is affecting also waking up of shoot cluster that was hibernated since long time.

```bash
$ k logs kube-apiserver-5cc67cf8fd-wdnkk kube-apiserver
Error: invalid argument "" for "--advertise-address" flag: failed to parse IP: ""
Usage:
  hyperkube kube-apiserver [flags]
```

After some time, the `kube-apiserver` deployment has been properly configured with `--advertise-address`.

**Comment by ialidzhikov** [comment]
> After some time, the kube-apiserver deployment has been properly configured with --advertise-address.

What it means after some time?

**Comment by vpnachev** [comment]
5-10 minutes later. I believe something wrong has happened with SNI because currently the `--advertise-address` flag is set to the ClusterIP of the service which is of type LoadBalancer  and this still so after several reconciliations and more than 1 hour later :open_mouth:.

cc @mvladev

**Comment by mvladev** [comment]
I'm on it.

**Comment by mvladev** [comment]
Have a look at #2895

**Comment by vpnachev** [comment]
I have just faced this issue with a freshly created cluster. 
In my landscape there was no shoot clusters and the APIServerSNI feature gate is enabled.
I create a shoot cluster and its kube-apiserver service is of type LoadBalancer, this should not happen because this cluster never ever existed before.

**Comment by mvladev** [comment]
> I have just faced this issue with a freshly created cluster.
> In my landscape there was no shoot clusters and the APIServerSNI feature gate is enabled.
> I create a shoot cluster and its kube-apiserver service is of type LoadBalancer, this should not happen because this cluster never ever existed before.

This one is separate problem which is addressed in #2901
